### PR TITLE
Yogesh/feature/sp json logger

### PR DESCRIPTION
**Pull request by yogesh8177**
Changed logger from console to sp-json-logger

### CODE REVIEW

File: common/mixins/utils.js [hunk 1]
```diff
@@ -5,15 +5,16 @@ var Promise = require('bluebird');
...
 
 module.exports = function(Model, options) {
 
   Model.getCurrentUserModel = function(cb) {
     var ctx = loopback.getCurrentContext();
     var currentUser = ctx && ctx.get('currentUser');
     if (currentUser) {
-      log.trace('inside ' + Model.definition.name + '.getCurrentUserModel() - currentUser: ', currentUser.username);
+      //log.trace('inside ' + Model.definition.name + '.getCurrentUserModel() - currentUser: ', currentUser.username);
+      logger.trace({log: {message: `inside ${Model.definition.name}.getCurrentUserModel() - currentUser: ${currentUser.username}` }});
```
Comment (kamal0808) [inline common/mixins/utils.js]: Don't you think writing everything under `log.message` is a bit exhausting for all logs? 
There could be a utility function that could do this job on itself.
Something that would change:
`logger.trace('inside ${Model.definition.name}.getCurrentUserModel() - currentUser: ${currentUser.username}'); `
TO
`logger.trace({log: {message: 'inside ${Model.definition.name}.getCurrentUserModel() - currentUser: ${currentUser.username}' }});`

Comment (pulkitsinghal) [reply common/mixins/utils.js]: So maybe when only one string argument is available, the logger can automap it to `{log: {message: <arg1>} }` and the rest of the times, we will use the object structure when writing the logs as developers ... thoughts?

Comment (pulkitsinghal) [reply common/mixins/utils.js]: `logger.<level>('my ' + arg1 + ' should be printed with '  + JSON.stringify(arg2));`
becomes:
`logger.<level>( { log: { message: 'my ' + arg1 + ' should be printed with '  + JSON.stringify(arg2) } } );`

Comment (pulkitsinghal) [reply common/mixins/utils.js]: and when its important to log differently, then the developers can choose to write themselves:
`logger.<level>( { log: { message: 'some reasoning',  arg1: arg1, arg2: arg2 } } );`

Comment (yogesh8177) [reply common/mixins/utils.js]: So maybe when only one string argument is available, the logger can automap it to {log: ```{message: <arg1>} }``` +1

```logger.<level>( { log: { message: 'my ' + arg1 + ' should be printed with ' + JSON.stringify(arg2) } } );``` 
Second example might create problem in logstash pipeline while parsing to json, and we will loose ability to index individual fields if we are stringifying json object inside message field! Should we sacrifice that for ease of use?

**Comment by kamal0808** [comment]
- Few files under common/models are still left. Best way to do is find `console` or `log` in path common/models
-

**Comment by kamal0808** [comment]
@yogesh8177 Where are we on this pull request? Were you able to complete it in rest of the files?

File: common/utils/vend.js [hunk 2]
```diff
@@ -612,7 +612,7 @@ var createStockOrderForVend = function(storeModelInstance, reportModelInstance){
 var markStockOrderAsSent = function(storeModelInstance, reportModelInstance){
   var storeConfigId = storeModelInstance.storeConfigModelToStoreModelId;
   //log.debug('markStockOrderAsSent()', 'storeConfigId: ' + storeConfigId);
-  logger.debug({log: {message: 'markStockOrderAsSent()', storeConfig: storeConfig }});
+  logger.debug({log: {message: 'markStockOrderAsSent()', storeConfig: storeConfigId }});
```
Comment (pulkitsinghal) [inline common/utils/vend.js]: @kamal0808 - this is an intentional change right?

**Comment by kamal0808** [comment]
Yes. Purely intentional. It is supposed to be `storeConfigId`

**Pull request merged**